File: go/config/UPnp.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// UPnp 对应 config/UPnp.py。
type UPnp struct {
	*session.ApiGroup
}

// NewUPnp 创建 UPnp API 分组。
func NewUPnp(s *session.Session) *UPnp {
	return &UPnp{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 upnp/config.xml。
func (u *UPnp) Config() (map[string]interface{}, error) {
	return u.S.Get("upnp/config.xml", nil, "config")
}

File: src/config/u_pnp.rs
//! UPnp config group (`config/UPnp.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// UPnP config group.
pub struct UPnPConfig<'a> {
    conn: &'a Connection,
}

impl<'a> UPnPConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        UPnPConfig { conn }
    }

    /// `upnp/config.xml`. UPnP configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/upnp/config.xml")
    }
}
